### ISSUE TITLE
Cannot install on new Raspberry Pi 

### ISSUE THREAD

**Issue by jplwoodward** [ispysoftware/agent-install-scripts#3]
Here is the result of your install script:

pi@Pi4:/etc $ sudo bash <(curl -s "https://raw.githubusercontent.com/ispysoftware/agent-install-scripts/main/install.sh")
bash: /dev/fd/63: No such file or directory
pi@Pi4:/etc $ (23) Failed writing body

NEVERMIND:  did not need the sudo

**Comment by jplwoodward** [comment]
I also cannot instal under Docker per your instructions:

docker run -it -p 8090:8090 -p 3478:3478/udp -p 50000-50010:50000-50010/udp \ -v /appdata/ispyagentdvr/config/:/agent/Media/XML/ \ -v /appdata/ispyagentdvr/media/:/agent/Media/WebServerRoot/Media/ \ -v /appdata/ispyagentdvr/commands:/agent/Commands/ \ -e TZ=America/Los_Angeles \ --name agentdvr doitandbedone/ispyagentdvr
docker: invalid reference format.

**Comment by jplwoodward** [comment]
When I installed in Docker I got this warning and it does not respond ob port 8090:

WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm/v7) and no specific platform was requested
standard_init_linux.go:228: exec user process caused: exec format error

I have the standard Raspberry Pi  OS Lite Bullseye image from Raspberry Pi Imager.  What am I doing wrong?

**Comment by jplwoodward** [comment]
I finally got the linux install script to at least rub, and said to install ffmpeg when asked.  Hoeever, it did not work:

pi@Pi4:~ $ ./start_agent.sh
Starting...
Creating temp path for opencv in /tmp/
Get com port list
Using default port (8090)
Local Server running on port 8090
H264 compatible version of ffmpeg is required to use H264 encoding. Switch to VP8 in settings if unable to record.
Accepting connections at http://*:8090/
Access locally at http://192.168.1.57:8090
Agent (v3.7.6.0)
Log start
Process is 32 bit (ARM)
Using WorkStation GC
Init Webrtc
Init FFmpeg
Looking for ffmpeg in /usr/local/lib
Not in there..
Looking for ffmpeg in /usr/lib/arm-linux-gnueabihf
ffmpeg path: /usr/lib/arm-linux-gnueabihf
Initializing devices
Error: Unable to load DLL 'avdevice.58 under /usr/lib/arm-linux-gnueabihf': The specified module could not be found.
ZeroConf: ZeroConf Service Started
Error:    at FFmpeg.AutoGen.ffmpeg.LoadLibrary(String libraryName, Boolean throwException)
   at FFmpeg.AutoGen.ffmpeg.<>c.<.cctor>b__6_0(String libraryName)
   at FFmpeg.AutoGen.ffmpeg.<>c.<.cctor>b__6_173()
   at FFmpeg.AutoGen.ffmpeg.avdevice_register_all()
   at CoreLogic.FFmpeg.Init()
   at CoreLogic.Threads.StartUp()
Unable to load DLL 'avdevice.58 under /usr/lib/arm-linux-gnueabihf': The specified module could not be found.
Install FFMPEG (v4) using the terminal. See the download page on ispyconnect.com for instructions.
Killed

At the end it said "Install FFMPEG (v4) using the terminal. See the download page on ispyconnect.com "for instructions.  I could not find the instructions.

**Comment by jplwoodward** [comment]
Also, during the installation of ffmpeg there were many many many errors similar to:

<command-line>: error: ‘x265_12bit_interp_4tap_vert_sp_64x32_neon’ was not declared in this scope; did you mean ‘x265_interp_4tap_vert_sp_64x32_neon’?

**Comment by jplwoodward** [comment]
I started over...rerflashed Raspberry Pi OS Lite. did "sudo apt install ffmpeg", then you install script, saying NO to installing ffmpeg.  That seemed to install, BUT, when going to <ip address>:8090, your gui starts, asked for the language, but then says Connecting... FOREVER.

**Comment by ispysoftware** [comment]
What's the output in the console?

**Comment by jplwoodward** [comment]
I access the Pi via ssh (it is headless), AgentDVR is running as a service.  There is no output to the terminal.  How do I access the console?

**Comment by jplwoodward** [comment]
I stopped the service (which WAS running)...and the GUI loaded enough to allow me to change the theme and access help, but otherwise just said connecting...

After stopping the service, I started it manually, with the result(note that it aborted):

pi@Pi4:~ $ ./start_agent.sh
Starting...
Creating temp path for opencv in /tmp/
Get com port list
Using default port (8090)
Local Server running on port 8090
H264 compatible version of ffmpeg is required to use H264 encoding. Switch to VP8 in settings if unable to record.
Accepting connections at http://*:8090/
Access locally at http://192.168.1.57:8090
Agent (v3.7.6.0)
Log start
Process is 32 bit (ARM)
Using WorkStation GC
Init Webrtc
Init FFmpeg
Looking for ffmpeg in /usr/local/lib
Not in there..
Looking for ffmpeg in /usr/lib/arm-linux-gnueabihf
ffmpeg path: /usr/lib/arm-linux-gnueabihf
Initializing devices
vdpau
ZeroConf: ZeroConf Service Started
vaapi
drm
opencl
Found 8 devices
List Devices: /dev/video18
call ffplay -nodisp -f v4l2 -list_formats all /dev/video18
[video4linux2,v4l2 @ 0xaa100630] not a video capture device.
List Devices: /dev/video12
call ffplay -nodisp -f v4l2 -list_formats all /dev/video12
[video4linux2,v4l2 @ 0xaa000630] not a video capture device.
List Devices: /dev/video11
call ffplay -nodisp -f v4l2 -list_formats all /dev/video11
[video4linux2,v4l2 @ 0xaa000630] not a video capture device.
List Devices: /dev/video10
call ffplay -nodisp -f v4l2 -list_formats all /dev/video10
[video4linux2,v4l2 @ 0xaa000630] not a video capture device.
List Devices: /dev/video16
call ffplay -nodisp -f v4l2 -list_formats all /dev/video16
[video4linux2,v4l2 @ 0xaa100630] not a video capture device.
List Devices: /dev/video15
call ffplay -nodisp -f v4l2 -list_formats all /dev/video15
[video4linux2,v4l2 @ 0xaa000630] ioctl(vidioc_g_input): inappropriate ioctl for device
List Devices: /dev/video14
call ffplay -nodisp -f v4l2 -list_formats all /dev/video14
[video4linux2,v4l2 @ 0xaa000630] ioctl(vidioc_g_input): inappropriate ioctl for device
List Devices: /dev/video13
call ffplay -nodisp -f v4l2 -list_formats all /dev/video13
[video4linux2,v4l2 @ 0xaa100630] not a video capture device.
call arecord -l
Error: Cuda Check: Cuda Check: The type initializer for 'Emgu.CV.Cuda.CudaInvoke' threw an exception.
Error: Cuda Check:    at Emgu.CV.Cuda.CudaInvoke.get_HasCuda()
   at CoreLogic.Threads.StartUp()
Error: Cuda Check: Cuda Check: The type initializer for 'Emgu.CV.CvInvoke' threw an exception.
Error: Cuda Check:    at Emgu.CV.CvInvoke.CheckLibraryLoaded()
   at Emgu.CV.Cuda.CudaInvoke..cctor()
Error: Cuda Check: Cuda Check: Unable to load shared library 'cvextern' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libcvextern: cannot open shared object file: No such file or directory
Error: Cuda Check:    at Emgu.CV.CvInvoke.RedirectError(CvErrorCallback errorHandler, IntPtr userdata, IntPtr prevUserdata)
   at Emgu.CV.CvInvoke..cctor()
Unable to load shared library 'cvextern' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libcvextern: cannot open shared object file: No such file or directory
Error: Load OpenCV: To debug this check the userguide at https://www.ispyconnect.com/userguide-agent-dvr.aspx#troubleshooting
Preloading Images
Turn Server accepting connections at 192.168.1.57:3478 using range 50000-50010)
Unhandled exception. System.TypeInitializationException: The type initializer for 'Emgu.CV.MatInvoke' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'Emgu.CV.CvInvoke' threw an exception.
 ---> System.DllNotFoundException: Unable to load shared library 'cvextern' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libcvextern: cannot open shared object file: No such file or directory
   at Emgu.CV.CvInvoke.RedirectError(CvErrorCallback errorHandler, IntPtr userdata, IntPtr prevUserdata)
   at Emgu.CV.CvInvoke..cctor()
   --- End of inner exception stack trace ---
   at Emgu.CV.CvInvoke.CheckLibraryLoaded()
   at Emgu.CV.MatInvoke..cctor()
   --- End of inner exception stack trace ---
   at Emgu.CV.MatInvoke.cveMatCreate()
   at CoreLogic.Utilities.Imaging.LoadStaticImages()
   at CoreLogic.Threads.StartUp()
   at AgentCore.Code.Initialize.Go()
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_1(Object state)
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
Aborted

**Comment by jplwoodward** [comment]
Looks like cvextern could not be found. I ran ldd as you suggested:

pi@Pi4:~/AgentDVR $ ldd libcvextern.so
	linux-vdso.so.1 (0xbecdd000)
	/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so => /usr/lib/arm-linux-gnueabihf/libarmmem-v7l.so (0xb59eb000)
	libz.so.1 => /lib/arm-linux-gnueabihf/libz.so.1 (0xb59c3000)
	liblapack.so.3 => /lib/arm-linux-gnueabihf/liblapack.so.3 (0xb5453000)
	libcblas.so.3 => not found
	libatlas.so.3 => not found
	libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0xb543f000)
	libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0xb5413000)
	librt.so.1 => /lib/arm-linux-gnueabihf/librt.so.1 (0xb53fb000)
	libstdc++.so.6 => /lib/arm-linux-gnueabihf/libstdc++.so.6 (0xb5273000)
	libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0xb5202000)
	libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0xb51d5000)
	libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0xb5081000)
	/lib/ld-linux-armhf.so.3 (0xb6f38000)
	libblas.so.3 => /lib/arm-linux-gnueabihf/libblas.so.3 (0xb501a000)
	libgfortran.so.5 => /lib/arm-linux-gnueabihf/libgfortran.so.5 (0xb4f2a000)

Now what?

**Comment by jplwoodward** [comment]
Also I want to remind you that the Docker install did not work either.

**Comment by ispysoftware** [comment]
Thank you for reminding. E of that it's much appreciated. It looks from ldd
output that some libraries are "not found". I'd start by installing those.
Linux/ pi set ups are way harder to automate than windows/ Mac as they all
have different base libraries installed.

**Comment by ispysoftware** [comment]
The Docker install doesn't work on arm only x86

**Comment by jplwoodward** [comment]
sudo apt-get install libatlas-base-dev  fixed it

**Comment by ispysoftware** [comment]
Just google the libraries eg  llibcblas.so.3
https://stackoverflow.com/questions/53347759/importerror-libcblas-so-3-cannot-open-shared-object-file-no-such-file-or-dire

Let me know what you needed to install and I'll get the script updated

**Comment by jplwoodward** [comment]
Thanks for your responsiveness.  Here is what added the missing libraries:

sudo apt-get install libatlas-base-dev

Also, recall that when I used the install ffmpeg option in your script, I got *tons* of errors of the form:

: error: ‘x265_12bit_interp_4tap_vert_sp_64x32_neon’ was not declared in this scope; did you mean ‘x265_interp_4tap_vert_sp_64x32_neon’?

Instead, I manually installed ffmpeg (without errors) before running your script with:

sudo apt install ffmpeg

Everything works fine now.  Thank you.
